fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogFragment.kt

```diff
@@ -71,10 +79,41 @@ class TranslationsDialogFragment : BottomSheetDialogFragment() {
             ),
         )
 
+        // Signalling user intention to translate
+        browserStore.dispatch(TranslationsAction.TranslateExpectedAction(args.sessionId))
+
         setContent {
+            val translateFromLanguages = browserStore.observeAsComposableState {
+                    state ->
+                state.findTab(args.sessionId)?.translationsState?.supportedLanguages?.fromLanguages
+            }.value
+
+            val translateToLanguages = browserStore.observeAsComposableState {
+                    state ->
+                state.findTab(args.sessionId)?.translationsState?.supportedLanguages?.toLanguages
+            }.value
+
+            val initialFromSelected = browserStore.observeAsComposableState {
+                    state ->
+                state.findTab(args.sessionId)?.translationsState?.translationEngineState
+                    ?.initialFromLanguage(translateFromLanguages)
+            }.value
+
+            val initialToSelected = browserStore.observeAsComposableState {
+                    state ->
+                state.findTab(args.sessionId)?.translationsState?.translationEngineState
+                    ?.initialToLanguage(translateToLanguages)
+            }.value
```

Comment:
I would consider combining these into one `translationsState` observable and getting/computing the subproperties you need from there.

Reply:
Much neater, updated!